crowbar_framework/app/models/chef_object.rb

```diff
@@ -51,9 +51,9 @@ def self.chef_escape(str)
     str.gsub("-:") { |c| '\\' + c }
   end
 
-  def export(name = nil)
-    name ||= self.respond_to?(:name) ? self.name : "unknown"
-    file   = Rails.root.join("db", "#{self.chef_type}_#{name}.json")
-    File.open(file, "w") { |f| f.write(self.to_json) }
+  def export(name = nil, obj = self)
+    name ||= obj.respond_to?(:name) ? obj.name : "unknown"
+    file   = Rails.root.join("db", "#{obj.chef_type}_#{name}.json")
+    File.open(file, "w") { |f| f.write(obj.to_json) }
```

Comment:
I must admit I find it a bit hacky to abuse ChefObject.export for exporting the new Proposal object. But I understand it might be required for now (as fixing this likely requires changing the logging barclamp).

I see a FIXME at https://github.com/crowbar/barclamp-crowbar/pull/1254/files#diff-7c1b3ff9bea4e2068fbd92ad5b03dcd4R4 for this, but I'd welcome another FIXME here (and possibly a new task in the EPIC for the rearchitecture?)